### PR TITLE
Upgrade amoc_rest with new openapi gens and enforce OTP27

### PR DESCRIPTION
**Pull request by NelsonVides**
To merge after https://github.com/esl/amoc_rest/pull/11

After integrating the new generated code, there's a couple of differences, mostly as we enforce OTP27:
- With `json`, we don't need any `jsx`/`jiffy`/other, and it's also the fastest. The generated code also allow us to distinguish between HTTP calls with `content-type: application/json` vs others, see the pattern-matching clauses on `accept_callback/4`.
- With `code:get_doc/1`, documentation chunks are now embedded in the module and can be extracted and read at runtime, hence we can rid of `docsh`.

### CODE REVIEW

File: src/rest_api/amoc_api_helpers_scenario_info.erl [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
          scenario_params/1,
-         get_edoc/1]).
+         get_documentation/1]).
 
--spec get_edoc(module()) -> binary().
-get_edoc(Scenario) ->
-    case docsh_lib:get_docs(Scenario) of
+-spec get_documentation(module()) -> binary().
+get_documentation(Scenario) ->
+    case code:get_doc(Scenario) of
```
Comment (NelsonVides) [inline src/rest_api/amoc_api_helpers_scenario_info.erl:17]: This is one of my favourite things of upgrading to OTP27, docs are embedded on the module now.

File: test/amoc_api_helper.erl [hunk 2]
```diff
@@ -55,7 +60,8 @@ patch(BaseUrl, Path, Body) ->
 
 -spec request(string(), binary(), binary()) ->
     {integer(), json()}.
-request(BaseUrl, Path, Method) -> request(BaseUrl, Path, Method, <<"">>).
+request(BaseUrl, Path, Method) ->
+    request(BaseUrl, Path, Method, <<"">>).
```
Comment (JanuszJakubiec) [inline test/amoc_api_helper.erl]:
```suggestion
    request(BaseUrl, Path, Method, <<>>).
```

File: test/amoc_api_helper.erl [hunk 3]
```diff
@@ -42,11 +45,13 @@ put(BaseUrl, Path, Body) ->
 
 -spec patch(string()) ->
     {integer(), json()}.
-patch(Path) -> patch(get_url(), Path, <<"">>).
+patch(Path) ->
+    patch(get_url(), Path, <<"">>).
```
Comment (JanuszJakubiec) [inline test/amoc_api_helper.erl]:
```suggestion
    patch(get_url(), Path, <<>>).
```

**Review by JanuszJakubiec — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**